### ISSUE TITLE
Create Builds for as many platforms and architectures as possible

### ISSUE THREAD

**Issue by aphelionz** [ipfs-rust/rust-ipfs#89]
Here is what go-ipfs has - we should try to match as closely as possible.

![image](https://user-images.githubusercontent.com/106148/76467600-550ce300-63c0-11ea-9597-d0f1244ac098.png)

**Comment by aphelionz** [comment]
Removing the `cid` and `sha512` checksums, the list becomes a lot less daunting. Also, I believe we can make the decision to _not_ support any 386 or 32 bit builds. Does that sound reasonable?

- [ ] go-ipfs-source.tar.gz
- [ ] go-ipfs_v0.4.23_darwin-386.tar.gz
- [ ] go-ipfs_v0.4.23_darwin-amd64.tar.gz
- [ ] go-ipfs_v0.4.23_freebsd-386.tar.gz
- [ ] go-ipfs_v0.4.23_freebsd-amd64.tar.gz
- [ ] go-ipfs_v0.4.23_freebsd-arm.tar.gz
- [ ] go-ipfs_v0.4.23_linux-386.tar.gz
- [ ] go-ipfs_v0.4.23_linux-amd64.tar.gz
- [ ] go-ipfs_v0.4.23_linux-arm.tar.gz
- [ ] go-ipfs_v0.4.23_linux-arm64.tar.gz
- [ ] go-ipfs_v0.4.23_windows-386.zip
- [ ] go-ipfs_v0.4.23_windows-amd64.zip
- [ ] Source code

**Comment by koivunej** [comment]
What was the target @rklaehn is building for?

No idea if freebsd could be supported at the moment (rust tier 2 support but no idea if mio supports? A closed issue exists which depends on an open issue on libc which has been open for years).

Is that darwin-386 the platform which was made much more difficult recently by apple and dropped in tiered support from 1 to 3? Yes it was: https://blog.rust-lang.org/2020/01/03/reducing-support-for-32-bit-apple-targets.html ... What Apple devices actually have 32bit intel processors?

**Comment by aphelionz** [comment]
I believe that was old PowerPC things which are, in computer years, elderly.

**Comment by koivunej** [comment]
I don't think it can be ppc because the demotion blog post talks about specifically i686.. Apparently they sold MacBookPros with late i686 chips: https://support.apple.com/en-us/HT201948. I can't say I remember there even being Intel Core processors which were 32-bit, perhaps they were only for Apple? :)

**Comment by aphelionz** [comment]
Capturing @rklaehn 's input on Matrix:

> x86 32 bit is probably not as important as arm 32 bit.
> 
> I think having sufficient variety in build targets to make sure that there are no showstoppers would be enough. If somebody e.g. needs some obscure arm flavor they would have to build it themselves anyway.

**Comment by koivunej** [comment]
Note https://github.com/rs-ipfs/rust-ipfs/pull/469 changes this so that we'll temporarily drop the mobile platforms and only build on major 3 (linux, windows, mac) given limited CI seconds. Open to restore these if need be, but we never did formulate any tests on these mobile platforms either.
